binaryaudit/dnf.py

```diff
@@ -117,21 +127,28 @@ def generate_abidiffs(key, source_dir, new_json_file, old_json_file, output_dir,
         for arg in cmd_supporting_args:
             command_list.append(arg)
         with open("output_file", "w") as output_file:
+            start_time = time.time()
             abipkgdiff, abipkgdiff_exit_code = run.run_command(command_list, None, output_file)
+            exec_time = time.time()-start_time
+            with rpmfile.open(old_main_rpm) as rpm:
+                name = rpm.headers.get('name').decode('utf-8')
+                old_version = rpm.headers.get('version').decode('utf-8')
+                old_release = rpm.headers.get('release').decode('utf-8')
+            with rpmfile.open(new_main_rpm) as rpm:
+                new_version = rpm.headers.get('version').decode('utf-8')
+                new_release = rpm.headers.get('release').decode('utf-8')
+            old_VR = old_version + "-" + old_release
+            new_VR = new_version + "-" + new_release
+            out = ""
             if abipkgdiff_exit_code != 0:
-                with rpmfile.open(old_main_rpm) as rpm:
-                    name = rpm.headers.get('name').decode('utf-8')
-                    old_version = rpm.headers.get('version').decode('utf-8')
-                    old_release = rpm.headers.get('release').decode('utf-8')
-                with rpmfile.open(new_main_rpm) as rpm:
-                    new_version = rpm.headers.get('version').decode('utf-8')
-                    new_release = rpm.headers.get('release').decode('utf-8')
-                print("Incompatibility found between " + name + "-" + old_version + "-" + old_release
-                      + " and " + name + "-" + new_version + "-" + new_release)
-                fileName = name + "__" + old_version + "-" + old_release + "__" + new_version + "-" + new_release + ".abidiff"
-                os.rename("output_file", output_dir + fileName)
+                print("Incompatibility found between " + name + "-" + old_VR + " and " + name + "-" + new_VR)
                 status = "fail"
-                # insert into db
+                fileName = name + "__" + old_VR + "__" + new_VR + ".abidiff"
+                os.rename("output_file", output_dir + fileName)
+                with open(output_dir + fileName) as f:
+                    out = f.read()
+        insert_db(db_conn, build_id, product_id, name, old_VR, new_VR, exec_time, status, out)
```

Comment:
@sthiru224  could the status be an outcome of `abicheck.diff_get_bit`?

Thanks